scripts/generate-wasm.sh

```diff
@@ -69,15 +69,15 @@ if [[ "$(docker images -q ${IMAGE}:${TAG} 2> /dev/null)" == "" ]]; then
   git checkout ${ENVOY_SHA}
 
   # Rebuild and push
-  cd api/wasm/cpp && docker build -t ${IMAGE}:${TAG} -f Dockerfile-sdk .
+  cd api/wasm/cpp && docker build -t ${WASM_SDK_IMAGE}:${TAG} -f Dockerfile-sdk .
   if [[ ${PUSH_DOCKER_IMAGE} == 1 ]]; then
-    docker push ${IMAGE}:${TAG} || "fail to push to gcr.io/istio-testing hub"
+    docker push ${WASM_SDK_IMAGE}:${TAG} || "fail to push to gcr.io/istio-testing hub"
   fi
 fi
 
 # Regenerate all wasm plugins and compare diffs
 # Tag image to v3, which is what used by all build wasm script.
-docker tag ${IMAGE}:${TAG} ${IMAGE}:v3
+docker tag ${WASM_SDK_IMAGE}:${TAG} ${WASM_SDK_IMAGE}:v3
```

Comment:
We should probably use istio/proxy version instead of v3. We may need to re-compile old Wasm modules from older releases for a fix.

Reply:
Yeah the tag leads to confusion here. The wasm sdk build script actually does retag: https://github.com/istio/proxy/blob/96399ff4f54adac9152a07a0674c5e40cfbb09c3/scripts/generate-wasm.sh#L80 and sdk image is actually versioned with envoy sha.

I've updated the script to use envoy sha as in wasm gen script. I think it would help to clarify the thing a bit.